### PR TITLE
Don't advertise the reserved slots. Fix #1808

### LINKED ISSUE #1808: Server shows 62/64 when there are reserved slots
That has a bad UX IMO. Should show 62/62 instead.

### PR DESCRIPTION
**Pull request by Learath2**
I think this is also fine for vanilla, but I can't get the vanilla client to work on mac to test it.

### CODE REVIEW

File: src/engine/server/server.cpp [hunk 1]
```diff
@@ -1585,9 +1585,9 @@ void CServer::SendServerInfo(const NETADDR *pAddr, int Token, int Type, bool Sen
 	}
 
 	ADD_INT(p, PlayerCount); // num players
-	ADD_INT(p, MaxClients-g_Config.m_SvSpectatorSlots); // max players
+	ADD_INT(p, maximum(MaxClients - g_Config.m_SvSpectatorSlots - g_Config.m_SvReservedSlots, PlayerCount)); // max players
```
Comment (heinrich5991) [inline src/engine/server/server.cpp]:
```suggestion
	ADD_INT(p, maximum(MaxClients - maximum(g_Config.m_SvSpectatorSlots, g_Config.m_SvReservedSlots), PlayerCount)); // max players
```

I think it's this way? Not sure. Drop `sv_spectator_slots`?

Comment (Learath2) [reply src/engine/server/server.cpp]: I'm not sure either, but the way these slots work is weird, imagine a server with 15 reserved slots and 1 spectator slot, now people without the password can join the server and take up the "reserved" slots.

Comment (heinrich5991) [reply src/engine/server/server.cpp]: Reserved slots only work for joining the server currently, not reserving a spot in the game. So it'd be correct to display 0/1 there, I guess.

**Comment by heinrich5991** [comment]
bors r+

Thanks.

**Pull request merged**